src/api/mod.rs

```diff
@@ -403,6 +451,119 @@ impl<'a> ValueListBuilder<'a> {
     }
 }
 
+//TODO: move to its own module metadata, with related types
+fn to_meta_data(meta_hm: &HashMap<String, MetaValue>) -> Result<*mut meta_data_t, SubmitError> {
```

Comment:
I believe this function signature can be changed to receive an iterator of key value tuples. HashMap functionality doesn't appear to be used. Not too big of an issue as this function is internal, but it would be one less change in case we need to switch the container from a hashmap to another structure (like vec)

Reply:
i've changed it to accept a type that implements IntoIterator<Item=(&string, &MetaValue)>.